lib/Listener/AlbumsManagementFileEventListener.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace OCA\Photos\Listener;
+
+use OCA\Photos\Album\AlbumMapper;
+use OCP\Files\File;
+use OCP\Files\Folder;
+use OCP\Files\Node;
+use OCP\EventDispatcher\Event;
+use OCP\EventDispatcher\IEventListener;
+use OCP\Files\Cache\CacheEntryRemovedEvent;
+use OCP\Share\Events\ShareDeletedEvent;
+use OCP\User\Events\UserDeletedEvent;
+
+class AlbumsManagementFileEventListener implements IEventListener {
+	private AlbumMapper $albumMapper;
+
+	public function __construct(AlbumMapper $albumMapper) {
+		$this->albumMapper = $albumMapper;
+	}
+
+	public function handle(Event $event): void {
+		if ($event instanceof CacheEntryRemovedEvent) {
+			// Remove node from all albums containing it.
+			$albums = $this->albumMapper->getForFile($event->getFileId());
+			foreach ($albums as $album) {
+				$this->albumMapper->removeFile($album->getId(), $event->getFileId());
+			}
+		}
+
+		if ($event instanceof UserDeletedEvent) {
+			// Delete all user's albums.
+			$albums = $this->albumMapper->getForUser($event->getUser()->getUID());
+			foreach ($albums as $album) {
+				$this->albumMapper->delete($album->getId());
+			}
+		}
+
+		if ($event instanceof ShareDeletedEvent) {
+			$receiverId = $event->getShare()->getSharedWith();
+			$this->forEachSubNode(
```

Comment:
Going through each subnodes of the share to remove them from albums.
Could be problematic performance wise, no ? Would it make sense to put that into a job ?

Reply:
Discussed, and an event will be emitted for each deleted entries by the server.